boards/risc-v/esp32c3/esp32c3-devkit/scripts/Make.defs

```diff
@@ -41,8 +41,8 @@ ifneq ($(CONFIG_DEBUG_NOOPT),y)
   ARCHOPTIMIZATION += $(MAXOPTIMIZATION) -fno-strict-aliasing
 endif
 
-ARCHCFLAGS = -fno-common -ffunction-sections -fdata-sections -msmall-data-limit=0
-ARCHCXXFLAGS = $(ARCHCFLAGS) -fno-exceptions -fcheck-new -fno-rtti
+ARCHCFLAGS += -fno-common -ffunction-sections -fdata-sections -msmall-data-limit=0
+ARCHCXXFLAGS += -fno-common -ffunction-sections -fdata-sections -msmall-data-limit=0 -std=c++17
```

Comment:
I was almost starting to do the opposite change and do `ARCHCXXFLAGS += $(ARCHCFLAGS) ....` for all the platforms as sometimes `-fno-common -ffunction-sections -fdata-sections` are not added to `ARCHCXXFLAGS`

Reply:
yes, it's better to add C/C++ shared option to ARCHCFLAGS and C++ specifical option to ARCHCXXFLAGS, so we can remove the unnecessary duplication.

Reply:
Do you have an example of "C-specific flags" that will not fit to C++ compiler input?

Reply:
Not from my mind, but I may be convinced I can find any documentation stating that the C++ frontend accepts every flag also accepted by the C frontend.

Reply:
yeah. this might be the case

Reply:
```bash
$ g++ -fallow-parameterless-variadic-functions ~/hello_main.c 
cc1plus: warning: command-line option ‘-fallow-parameterless-variadic-functions’ is valid for C/ObjC but not for C++
```